### ISSUE TITLE
Write API Documentation in Wiki

### ISSUE THREAD

**Issue by lokka30** [lokka30/Treasury#280]
Hey all,

Treasury has recently released it's Version 2 update and requires a full Wiki refresh. At the moment, all that is left to write is the API documentation. All of Treasury's APIs have javadocs, so writing the Wiki should be a trivial matter of explaining core concepts and giving examples on how to use the APIs for common requirements. Our core development team have spent a lot of time designing and writing the code and would greatly appreciate community contributions to finish the Wiki.

If you're interested in helping out, the Wiki pages have been set so anyone can edit them - please feel free to get started and leave a comment if you have any questions.

The following Wiki pages are outdated as of the recent API revisions in Treasury v2, and we wish to address them in this issue.

- [ ] [Economy API](https://github.com/ArcanePlugins/Treasury/wiki/Treasury's-Economy-API)
- [ ] [Services API](https://github.com/ArcanePlugins/Treasury/wiki/Treasury-Services-API)
- [ ] [Events API](https://github.com/ArcanePlugins/Treasury/wiki/Treasury-Events-API)

The first API documentation page I recommend we focus on at the moment is the [Economy API](https://github.com/ArcanePlugins/Treasury/wiki/Treasury's-Economy-API), which will be the most content-heavy of them all. This page will not only need to describe the API's features (as it currently does but for the previous API revision), but also *how* to use each feature. The same type of content and layout should follow with the other API pages.

Documentation contributions will be officially recorded in the Credits page and this may be useful if you would like to contribute to open-source projects for your resume.

Thanks!

**Comment by Tarlsie** [comment]
Hi, I would like to help with the API documentation.

**Comment by lokka30** [comment]
> Hi, I would like to help with the API documentation.

Hi @Tarlsie, hope you're well and thank you for helping us with the documentation. 🙂 

I've edited the original post to include some more useful information on what we need to work on - but before we start, it'd be worth me creating a document structure so we can fill in the blanks for each API's page. I'll action this.

Please confirm if GitHub allows you to contribute changes directly to the Wiki, or would contributions require cloning the [Wiki's repository](https://github.com/ArcanePlugins/Treasury.wiki.git)?

**Comment by LcyDev** [comment]
I want to help out too.

**Comment by lokka30** [comment]
> I want to help out too.

Thanks for your interest @LcyDev! Does the GitHub interface allow you to make edits to the Wiki at all, or will this require raw edits to the Markdown?

**Comment by Psychodahlia** [comment]
Hi @lokka30 if you still need help I would be happy to contribute to the API documentation. I have access to Postman, and I've taught myself API documentation. I'm looking for some hands-on experience. 

I clicked through the wiki and it's not letting me edit.

**Comment by lokka30** [comment]
> Hi @lokka30 if you still need help I would be happy to contribute to the API documentation. I have access to Postman, and I've taught myself API documentation. I'm looking for some hands-on experience.
> 
> I clicked through the wiki and it's not letting me edit.

Hi @Psychodahlia, thanks for your interest! I've just spotted a setting that allows all users to edit the Wiki and adjusted accordingly. @Tarlsie and @LcyDev - hopefully this helps.

Cheers

**Comment by LcyDev** [comment]
Okay, it seems I can edit the wiki just fine.

I can't seem to access the [JavaDocs](https://arcaneplugins.github.io/Treasury/javadocs/)
Is there an issue?

**Comment by lokka30** [comment]
> Okay, it seems I can edit the wiki just fine.
> 
> I can't seem to access the [JavaDocs](https://arcaneplugins.github.io/Treasury/javadocs/) Is there an issue?

Hey @LcyDev, it's possible there was an issue with Ivan's server at the time though it seems to work for me fine just now: [Javadocs](https://jd.mrivanplays.com/treasury/api/index.html)

**Comment by Chizobaonorh** [comment]
Hello, I would love to help out in this documentation.

**Comment by lokka30** [comment]
Hi @Chizobaonorh, thank you very much for your interest. I've updated the original post with further details. Feel free to make edits to the Wiki, and let us know if you have any questions :)

**Comment by Chizobaonorh** [comment]
My apologies for the late reply, i should solely focus on the economy wiki for now right?

**Comment by LcyDev** [comment]
Can we make contact in discord so we can be more on touch and sincronized?

**Comment by lokka30** [comment]
Absolutely, [here's a direct invite to the Treasury Discord channel](https://discord.gg/arcaneplugins-752310043214479462) on ArcanePlugins :+1:

**Comment by EdwinChukwuebuka** [comment]
Hello, I'd like to also help with the documentation

**Comment by lokka30** [comment]
> Hello, I'd like to also help with the documentation

Hi @EdwinChukwuebuka, thanks! If the original post doesn't have enough info please comment or shoot me a DM :)
Otherwise feel free to contribute as you wish, the Wiki should have public editing enabled.

**Comment by EdwinChukwuebuka** [comment]
Okay @lokka30 
I will, thank you :)

**Comment by EdwinChukwuebuka** [comment]
Okay @lokka30 
I will, thank you :)

**Comment by nimraahmed** [comment]
Hello, I'd like to collaborate and contribute to this documentation
